src/renderers/WebGLRenderer.js

```diff
@@ -1299,6 +1331,37 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 	}
 
+	function objectCanBeVisible( object ) {
```

Comment:
I agree that the name is improvable. There are those `.visible` properties on objects and materials, so "visible" can mean a lot of things, and just because an object's bounding geometry intersects the frustum doesn't mean it's necessarily visible.

Something like `isNotCulled`, `isObjectNotCulled` or `objectIsNotCulled` might be even better. You like?

Note that one *could* make it public and call from `WebGLShadowMap` but I intentionally did not do this because:
- There are only clipping planes when `renderer.clipShadows === true`
- in this case, I interpret clipping as a modeling operation (in line with the typical use cases) - we're cutting objects, not visible space.

BTW (off-topic) https://2016.revision-party.net/  (Shader Live Coding Finale right now)